src/environmentd/tests/sql.rs

```diff
@@ -1100,6 +1104,61 @@ source materialize.public.t1 (u1, storage):
     assert_eq!(explain, expect, "{explain}\n\n{expect}");
 }
 
+// Test `EXPLAIN TIMESTAMP AS JSON`
+#[test]
+fn test_explain_timestamp_json() {
+    let config = util::Config::default();
+    let server = util::start_server(config).unwrap();
+    let mut client = server.connect(postgres::NoTls).unwrap();
+    client.batch_execute("CREATE TABLE t1 (i1 int)").unwrap();
+
+    let row = client
+        .query_one("EXPLAIN TIMESTAMP AS JSON FOR SELECT * FROM t1;", &[])
+        .unwrap();
+    let explain: String = row.get(0);
+    // Just check that we can round-trip to the original type
+    let _explain: TimestampExplanation<Timestamp> = serde_json::from_str(&explain).unwrap();
+}
+
+// Test that the since for `mz_cluster_replica_utilization` is held back by at least
+// 30 days, which is required for the frontend observability work.
+//
+// Feel free to modify this test if that product requirement changes,
+// but please at least keep _something_ that tests that custom compaction windows are working.
+#[test]
+fn test_utilization_hold() {
+    let config = util::Config::default();
+    let server = util::start_server(config).unwrap();
+    let mut client = server.connect(postgres::NoTls).unwrap();
+    let q =
+        "EXPLAIN TIMESTAMP AS JSON FOR SELECT * FROM mz_internal.mz_cluster_replica_utilization";
+    let row = client.query_one(q, &[]).unwrap();
+    let explain: String = row.get(0);
+    let explain: TimestampExplanation<Timestamp> = serde_json::from_str(&explain).unwrap();
+    // If we're not in EpochMilliseconds, the timestamp math below is invalid, so assert that here.
+    assert!(matches!(
+        explain.determination.timestamp_context,
+        TimestampContext::TimelineTimestamp(Timeline::EpochMilliseconds, _)
+    ));
+    let since = explain
+        .determination
+        .since
+        .into_option()
+        .expect("The since must be finite");
+    const THIRTY_DAYS_MS: u64 = 30 * 24 * 60 * 60 * 1000;
+    let since_plus_30d = since.step_forward_by(&Timestamp::from(THIRTY_DAYS_MS));
+    let now = Timestamp::from(
+        u64::try_from(
+            SystemTime::now()
+                .duration_since(UNIX_EPOCH)
+                .unwrap()
+                .as_millis(),
+        )
+        .unwrap(),
+    );
+    assert!(since_plus_30d.less_equal(&now));
```

Comment:
As discussed in person, It seems like this is implying that when we create a table we set it's `since` to some time 30 days in the past which I think is potentially wrong.

This is existing behavior, but we probably shouldn't encode it in this test. Instead we should use deterministic timestamps to create a table and then fast forward 30 days. Take a look at `test_timestamp_recovery` for an example.